Create an npm script for starting app services standalone, and make their lifetime configurable

victorporof opened the pull request:
Fixes https://github.com/mozilla/tofino/issues/607

@Mossop

victorporof commented:
@jsantell you've been working on UAS stuff recently too, might want to be aware of this change.

jsantell commented on app/main/spawn.js:
```diff
@@ -13,38 +13,60 @@ specific language governing permissions and limitations under the License.
-  });
+    // Careful passing in options.verbose as true, as this stops the services from
+    // outliving the their original parents, which we want to allow in the UAS case.
+    stdio: options.verbose ? ['ignore', process.stdout, process.stderr] : ['ignore'],
```
[This breaks Windows](e8047ffe146e8cd9bc586b4cefc11ad546b91ca2) (should add a comment w/r/t this)

victorporof replied:
You mean in addition to the comment above?

jsantell replied:
The stdio options breaks Windows according to @mossop which was already reverted in the linked commit, so we should comment on that, and these changes would break Windows

jsantell commented:
What's the use-case for this? Couldn't we just use the [services CLI directly](https://github.com/mozilla/tofino/blob/master/app/services/user-agent-service/README.md)?

jsantell commented on app/main/spawn.js:
```diff
@@ -13,38 +13,60 @@ specific language governing permissions and limitations under the License.
-  process.on('exit', () => {
-    child.kill('SIGKILL');
-  });
+function killChildren(code) {
```
We can probably have a better function name for this.

victorporof replied:
See #607. The CLI doesn't start all services, and doesn't address the lifetime issues.

jsantell replied:
I just mean a better name than `killChildren`

victorporof replied:
This is in reply to your "What's the use-case for this?" question.

victorporof commented:
Addressed comments and changed wording to make it clear that we do want to run the servers attached to their parent process on npm run serve. @jsantell, this doesn't break anything on windows, and I don't see any further reason for your concerns.
